framework/src/packets/checksum.rs

```diff
@@ -1,8 +1,115 @@
 use common::Result;
-use packets::ip::IpAddrMismatchError;
-use std::net::IpAddr;
+use packets::ip::{IpAddrMismatchError, ProtocolNumber};
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 use std::slice;
 
+/// Generic Pseudo Header used for checksumming purposes.
+pub enum PseudoHeader {
+    V4 {
+        src: Ipv4Addr,
+        dst: Ipv4Addr,
+        packet_len: u16,
+        protocol: ProtocolNumber,
+    },
+    V6 {
+        src: Ipv6Addr,
+        dst: Ipv6Addr,
+        packet_len: u16,
+        protocol: ProtocolNumber,
+    },
+}
+
+impl PseudoHeader {
+    pub fn sum(&self) -> u16 {
+        let mut sum = match *self {
+            PseudoHeader::V4 {
```

Comment:
@drunkirishcoder did want to pass structs right through here to the v4/v6 funs, but not yet doable as you gotta destructure first, but I wanted the enum variant lookup in one place.

Reply:
I think this is good.